src/os/ui/filter/im/filterimport.js

```diff
@@ -208,151 +216,24 @@ os.ui.filter.im.FilterImportCtrl.prototype.init_ = function() {
  * @protected
  */
 os.ui.filter.im.FilterImportCtrl.prototype.onImportComplete = function(event) {
-  var importer = /** @type {os.im.Importer} */ (event.target);
-  var filters = /** @type {Array<!os.filter.FilterEntry>} */ (importer.getData());
-  importer.dispose();
-
-  var matched = {};
-  var unmatched = [];
-  var matchedCount = 0;
-  var layerId = /** @type {string} */ (this.scope['layerId']);
-
-  for (var i = 0, ii = filters.length; i < ii; i++) {
-    var filter = filters[i];
-    var filterTitle = filter.getTitle();
-    var typeOrFilterKey = filter.getType();
-    var tooltip = this.getFilterTooltip(filter);
-    var icons;
-    var layerTitle;
-    var filterModel;
-    var layerModel;
-
-    if (layerId) {
-      // if we have a layer ID, we were passed some context from a filter window, so use it
-      var impliedFilterable = os.ui.filter.getFilterableByType(layerId);
-      var columns = impliedFilterable && impliedFilterable.getFilterColumns();
-
-      if (impliedFilterable && columns && filter.matches(columns)) {
-        // this filter matches the columns of the passed in context, so add it as such
-        var clone = filter.clone();
-        clone.setId(goog.string.getRandomString());
-        clone.setType(layerId);
-
-        filterModel = this.getFilterModel(filterTitle, clone, tooltip);
-
-        if (matched[layerId]) {
-          // add to the existing layer item
-          matched[layerId]['filterModels'].push(filterModel);
-        } else {
-          // define a new layer item
-          icons = this.getIconsFromFilterable(impliedFilterable);
-          layerTitle = this.getTitleFromFilterable(impliedFilterable, layerId);
-          layerModel = this.getLayerModel(layerTitle, icons, clone.getMatch(), filterModel);
-          matched[layerId] = layerModel;
-        }
-
-        matchedCount = os.ui.filter.im.FilterImportCtrl.getFilterCount(filterModel, matchedCount);
-      }
-    } else {
-      var filterableTypes = os.ui.filter.getFilterableTypes(typeOrFilterKey);
-      var filterables = filterableTypes.map(os.ui.filter.getFilterableByType);
-
-      filterables.forEach(function(filterable) {
-        if (filterable) {
-          // we matched it by filter key, so clone the filter and set the internal filterable ID as its type
-          var clone = filter.clone();
-          var type = filterable.getFilterableTypes()[0];
-          clone.setId(goog.string.getRandomString());
-          clone.setType(type);
+  if (this.importer) {
+    // assign all the display values
+    this['matched'] = this.importer.matched;
+    this['unmatched'] = this.importer.unmatched;
```

Comment:
The `unmatched` property was originally an Array on the `FilterImportCtrl`, but the importer now has it as an Object. Is that intentional?

Reply:
Ah, good call that was not intentional and has been fixed.